app/src/main/java/com/github/se/eduverse/repository/ProfileRepository.kt

```diff
@@ -460,4 +464,62 @@ open class ProfileRepositoryImpl(
       emptyList()
     }
   }
+
+  override suspend fun deletePublication(publicationId: String, userId: String): Boolean {
+    return try {
+      coroutineScope {
+        val pubQuery =
+            publicationsCollection
+                .whereEqualTo("id", publicationId)
+                .whereEqualTo("userId", userId)
+                .get()
+                .await()
+
+        if (pubQuery.isEmpty) {
+          throw Exception("Publication not found or user not authorized")
+        }
+
+        val pubDoc = pubQuery.documents[0]
+        val mediaUrlToDelete = pubDoc.getString("mediaUrl")
+        val thumbnailUrlToDelete = pubDoc.getString("thumbnailUrl")
+        val likedByUsers = pubDoc.get("likedBy") as? List<String> ?: emptyList()
+
+        // Delete Firestore document first
+        pubDoc.reference.delete().await()
+
+        // Use async for parallel deletion of liked publications
+        val likedPubsDeletion = launch {
+          likedByUsers.forEach { likedUserId ->
+            val likedPubRef =
+                usersCollection
+                    .document(likedUserId)
+                    .collection("likedPublications")
+                    .document(publicationId)
+            likedPubRef.delete().await()
+          }
+        }
+
+        // Delete media files in parallel
+        val mediaDeletion = launch {
+          if (!mediaUrlToDelete.isNullOrEmpty()) {
+            val mediaRef = storage.getReferenceFromUrl(mediaUrlToDelete)
+            mediaRef.delete().await()
+          }
+
+          if (!thumbnailUrlToDelete.isNullOrEmpty() && thumbnailUrlToDelete != mediaUrlToDelete) {
+            val thumbnailRef = storage.getReferenceFromUrl(thumbnailUrlToDelete)
+            thumbnailRef.delete().await()
+          }
+        }
+
+        // Wait for all deletions to complete
+        likedPubsDeletion.join()
+        mediaDeletion.join()
```

Comment:
I forgot to save the comment in my review, so this are the 2 lines I think are not needed